src/community/schemaless-features/mongodb-schemaless/src/main/java/org/geoserver/schemalessfeatures/mongodb/mappers/SchemalessMongoToComplexMapper.java

```diff
@@ -73,6 +82,23 @@ public Feature buildFeature(DBObject rootDBO) {
         return f;
     }
 
+    private void handleGeometryReprojection(GeometryAttribute geom) {
+        if (crs == null) return;
+        Geometry geomValue = (Geometry) geom.getValue();
+        // if crs is not wgs84, we need to reproject the geometry
+        if (!CRS.equalsIgnoreMetadata(DefaultGeographicCRS.WGS84, crs)) {
+            try {
+                geomValue =
+                        JTS.transform(
```

Comment:
Repeatedly checking equality and looking up the transformation is not ideal, it may both slow down the operation and reduce scalability (lookups are synchronized as far as I remember). So do it once in the constructor, and here check if you have the MathTransform field populated, or not, instead